backend/src/main/java/com/huellapositiva/domain/actions/UpdateVolunteerProfileAction.java

```diff
@@ -0,0 +1,140 @@
+        jpaVolunteer.setInstagram(profileDto.getInstagram());
+        jpaVolunteer.setLinkedin(profileDto.getLinkedin());
+        jpaVolunteer.setAdditionalInformation(profileDto.getAdditionalInformation());
+    }
+
+    /**
+     * This method update information in credential table
+     *
+     * @param profileDto New user credential information to update
+     * @param jpaVolunteer The information in Database
+     */
+    private void updateCredentials(ProfileDto profileDto, JpaVolunteer jpaVolunteer) {
+        jpaVolunteer.getCredential().setName(profileDto.getName());
+        jpaVolunteer.getCredential().setSurname(profileDto.getSurname());
+        jpaVolunteer.getCredential().setEmail(profileDto.getEmail());
+        jpaVolunteer.getCredential().setBirthDate(parseToLocalDate(profileDto.getBirthDate()));
+        jpaVolunteer.getCredential().setPhoneNumber(profileDto.getPhoneNumber());
+    }
+
+    /**
+     * This method update information in credential table
+     *
+     * @param profileDto New user credential information to update
+     * @param email Email of user logged
+     */
+    private JpaLocation updateLocation(ProfileDto profileDto, String email) {
+        JpaVolunteer jpaVolunteer = jpaVolunteerRepository.findByEmailProfileInformation(email);
+        String id;
+        if (jpaVolunteer.getLocation() == null) {
+            id = Id.newId().toString();
+        } else {
+            id = jpaVolunteer.getLocation().getId();
+        }
+        return JpaLocation.builder()
+                .id(id)
+                .province(profileDto.getProvince())
+                .town(profileDto.getTown())
+                .address(profileDto.getAddress())
+                .island(profileDto.getIsland())
+                .zipCode(profileDto.getZipCode()).build();
+    }
+
+    /**
+     * This method checks if the credential information provided is not null
+     *
+     * @param profile The information to check
+     */
+    private boolean someFieldIsEmptyCredentials(ProfileDto profile) {
+        return profile.getName() == null
+                || profile.getSurname() == null
+                || profile.getBirthDate() == null
+                || profile.getEmail() == null
+                || profile.getPhoneNumber() == null;
+    }
+
+    /**
+     * This method checks if the island and zip code is not null
+     *
+     * @param profileDto The information to check
+     */
+    private boolean someFieldEmptyLocation(ProfileDto profileDto) {
+        return profileDto.getIsland() == null || profileDto.getZipCode() == null;
+    }
+
+    /**
+     * This method parse the Date in String to LocalDate
+     *
+     * @param date This data comes from the profile
+     */
+    public LocalDate parseToLocalDate(String date){
+        String[] parts = date.split("-");
+        return LocalDate.of(Integer.parseInt(parts[0]),Integer.parseInt(parts[1]),Integer.parseInt(parts[2]));
+    }
```

Comment:
Configure common mapper and customise date serialisation/deserialisation

Reply:
It's done